tests/unit/data/gomod-mocks/generate.sh

```diff
@@ -0,0 +1,27 @@
+#!/bin/bash
+set -o errexit -o nounset -o pipefail
+
+git clone https://github.com/cachito-testing/gomod-pandemonium
+trap 'rm -rf gomod-pandemonium' EXIT
+
+echo "---------------------------------------------------------------"
+(
+    # cd in a subshell, doesn't change the $PWD of the main process
+    cd gomod-pandemonium
+
+    echo "generating go_mod_download.json"
+    go mod download -json > ../go_mod_download.json
+
+    echo "generating go_list_deps_all.json"
+    go list -deps -json=ImportPath,Module,Standard,Deps all > ../go_list_deps_all.json
+
+    echo "generating go_list_deps_threedot.json"
+    go list -deps -json=ImportPath,Module,Standard,Deps ./... > ../go_list_deps_threedot.json
+
+    echo "generating vendor_modules.txt"
+    go mod vendor
+    cp vendor/modules.txt ../vendor_modules.txt
+)
+echo "---------------------------------------------------------------"
+
+echo "please adjust resolve_gomod_output[_vendored].json accordingly"
```

Comment:
Took me a minute to figure out what this meant. Different from the integration tests, this generates data that will be used during the test, not the final output that will be used in assertions.

Maybe add a header comment to explain that?

> The generated data will be used as mock output for Go commands, which are called internally by Cachi2. This will may pontentially change the assertion data located in resolve_gomod_output[_vendored].json. The content of these files needs to be manually adjusted.

Reply:
Restructured the script a bit, now the message about having to edit the expected data is only printed if there's a git diff in the mocked data